### PR TITLE
refactor/all: refactor mock-vault based on lastest safe-nd changes

### PR DESCRIPTION
**Pull request by Yoga07**

### CODE REVIEW

File: safe_core/src/client/mock/vault.rs [hunk 1]
```diff
@@ -339,6 +340,114 @@ impl Vault {
         }
     }
 
+    fn check_pub_perms_adata(
```
Comment (lionel1704) [inline safe_core/src/client/mock/vault.rs]: This function can be clubbed with `check_unpub_perms`

File: safe_core/src/client/mock/vault.rs [hunk 2]
```diff
@@ -339,6 +340,114 @@ impl Vault {
...
+                        Err(SndError::AccessDenied)
+                    }
+                },
+            ),
+            _ => Err(SndError::InvalidOperation),
+        }
+    }
+
+    fn check_unpub_perms_adata(
+        &self,
```
Comment (lionel1704) [inline safe_core/src/client/mock/vault.rs]: Do you need the `self` parameter?

File: safe_core/src/client/mock/vault.rs [hunk 3]
```diff
@@ -339,6 +340,114 @@ impl Vault {
...
+                        }
+                    },
+                )
+            }
+            _ => Err(SndError::InvalidOperation),
+        }
+    }
+
+    fn check_perms_mdata(
+        &self,
```
Comment (lionel1704) [inline safe_core/src/client/mock/vault.rs]: Here too. Is this needed?

File: safe_core/src/client/mock/vault.rs [hunk 4]
```diff
@@ -1221,9 +1314,7 @@ impl Vault {
             message_id,
         })
     }
-    //
```
Comment (lionel1704) [inline safe_core/src/client/mock/vault.rs]: Also was this removed on purpose?

File: safe_core/src/client/mock/vault.rs [hunk 5]
```diff
@@ -954,60 +1063,54 @@ impl Vault {
...
-                        };
-                        match address {
-                            ADataAddress::PubSeq { .. } => {
-                                let res = match data {
-                                    AData::PubSeq(adata) => {
-                                        match adata.fetch_permissions_at_index(idx as u64) {
-                                            Some(perm) => Ok(perm.clone()),
-                                            None => Err(SndError::NoSuchEntry),
-                                        }
+                    .and_then(move |data| match address {
```
Comment (m-cat) [inline safe_core/src/client/mock/vault.rs]: It's not a part of your PR, but I don't understand the `Macro cannot be used here` comment a few lines up, above `address,`. This comment seems meaningless to me, can it be removed?

File: safe_core/src/client/mock/vault.rs (hunk 4 above)
Comment (Yoga07) [reply safe_core/src/client/mock/vault.rs]: This did not make much sense as there was only one function under AData, and the rest of the types did not have these kinda banners too :)

File: safe_app/src/tests/mod.rs [hunk 6]
```diff
@@ -326,15 +326,16 @@ fn unregistered_client() {
...
+                        assert_eq!(data.tag(), pub_adata.tag());
                     })
             })
             .then(move |_| {
                 client3
                     .get_adata(ADataAddress::UnpubUnseq { name: addr, tag })
                     .then(|res| {
                         match res {
-                            Err(CoreError::NewRoutingClientError(SndError::AccessDenied)) => (),
+                            Err(CoreError::NewRoutingClientError(SndError::NoSuchData)) => (),
```
Comment (lionel1704) [inline safe_app/src/tests/mod.rs]: This error type should not be changed.
If this error type is returned from safe_nd we can use `map_err` to return the error we need.

**Pull request merged**